pombola/hansard/views.py

```diff
@@ -60,40 +60,48 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class SittingView(DetailView):
-    model = Sitting
+START_TIME_RE = re.compile(
+    r'^(?P<start_date>\d{4}-\d{2}-\d{2})(?:-(?P<start_time>.*))?$'
+)
 
-    def get_object(self):
-        """Get the object based on venue and start date and time"""
 
-        venue_slug = self.kwargs['venue_slug']
-        start_date_and_time = self.kwargs['start_date_and_time']
+def get_sittings_from_slugs(venue_slug, start_date_and_time_from_url):
+    query_args = {
+        'venue__slug': venue_slug,
+    }
 
-        query_args = {
-            'venue__slug':     venue_slug,
-        }
+    m = START_TIME_RE.search(start_date_and_time_from_url)
+    if not m:
+        msg = "Malformed sitting date and start time '{0}'"
+        raise Http404(msg.format(start_date_and_time_from_url))
 
-        # add blank HH-MM-SS if needed
-        start_time_parts = re.split('-', start_date_and_time)
-        if len(start_time_parts) == 3:
-            start_time_parts.extend( [ '00','00','00' ] )
+    start_date, start_time = m.groups()
+    if start_time:
+        start_time = start_time.replace('-', ':')
 
-        # split the date and time up into the relevant fields
-        (year,month,day,hour,minute,second) = [ int(x) for x in start_time_parts ]
+    query_args['start_date'] = start_date
+    query_args['start_time'] = start_time
```

Comment:
Is it OK to pass plain strings like this to `.filter` when the actual field is a date? I couldn't see anything in the docs but I might well have just been looking in the wrong place!